torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -27,12 +27,14 @@ static std::unordered_map<std::string, ParameterType> type_map = {
     {"Scalar", ParameterType::SCALAR},
     {"int64_t", ParameterType::INT64},
     {"SymInt", ParameterType::SYM_INT},
+    {"SymFloat", ParameterType::SYM_FLOAT},
     {"double", ParameterType::DOUBLE},
     {"complex", ParameterType::COMPLEX},
     {"TensorList", ParameterType::TENSOR_LIST},
     {"c10::List<::std::optional<Tensor>>", ParameterType::TENSOR_LIST},
     {"IntArrayRef", ParameterType::INT_LIST},
     {"SymIntArrayRef", ParameterType::SYM_INT_LIST},
+    {"SymFloatArrayRef", ParameterType::SYM_FLOAT_LIST},
```

Comment:
All of this code for SymFloat and SymFloatArrayRef isn't actually doing anything, because you are never code generating SymFloat/SymFloatArrayRef entries in the string passed to Python argument parser.  You can drop all of it, you only need to change the semantics of `is_float_or_complex_list` to also accept SymFloat, and ditto with THPUtils_checkDouble to switch to is_float_or_symfloat.

Reply:
Should we do the same thing for SymInt? If not, what's the difference?